Lib/fontmake/instantiator.py

```diff
@@ -0,0 +1,499 @@
+        if instance.familyName:
+            font.info.familyName = instance.familyName
+        if instance.styleName:
+            font.info.styleName = instance.styleName
+        if instance.postScriptFontName:
+            font.info.postscriptFontName = instance.postScriptFontName
+        if instance.styleMapFamilyName:
+            font.info.styleMapFamilyName = instance.styleMapFamilyName
+        if instance.styleMapStyleName:
+            font.info.styleMapStyleName = instance.styleMapStyleName
+
+        # If the masters haven't set the OS/2 weight and width class, use the
+        # user-space values ("input") of the axis mapping in the Designspace file for
+        # weight and width axes, if they exist. Clamp the values to the valid ranges.
+        if info_instance.openTypeOS2WeightClass is None:
+            if "wght" in self.weight_width_axes:
+                weight_axis = self.weight_width_axes["wght"]
+                font.info.openTypeOS2WeightClass = weight_class_from_wght_value(
+                    weight_axis.map_backward(location[weight_axis.name])
+                )
+        if info_instance.openTypeOS2WidthClass is None:
+            if "wdth" in self.weight_width_axes:
+                width_axis = self.weight_width_axes["wdth"]
+                font.info.openTypeOS2WidthClass = width_class_from_wdth_value(
+                    width_axis.map_backward(location[width_axis.name])
+                )
+
+        # Glyphs
+        for glyph_name, glyph_mutator in self.glyph_mutators.items():
+            glyph = font.newGlyph(glyph_name)
+
+            try:
+                glyph_instance = glyph_mutator.instance_at(location_normalized)
+            except Exception as e:
+                raise RuntimeError(
+                    f"Failed to generate instance of glyph '{glyph_name}'."
+                ) from e
+            if self.round_geometry:
+                glyph_instance = glyph_instance.round()
+
+            # onlyGeometry=True does not set name and unicodes, in ufoLib2 we can't
+            # modify a glyph's name. Copy unicodes from default font.
+            glyph_instance.extractGlyph(glyph, onlyGeometry=True)
+            glyph.unicodes = self.glyph_name_to_unicodes[glyph_name]
+
+        # Process rules
+        glyph_names_list = self.glyph_mutators.keys()
+        resultNames = designspaceLib.processRules(
+            self.designspace_rules, location, glyph_names_list
+        )
+        for oldName, newName in zip(glyph_names_list, resultNames):
+            if oldName != newName:
+                swapGlyphNames(font, oldName, newName)
+
+        # Write _design_ location to instance's lib.
+        font.lib["designspace.location"] = list(location.items())
+
+        return font
+
+
+def anisotropic(location: Location) -> bool:
+    for v in location.values():
+        if isinstance(v, tuple):
+            return True
+    return False
+
+
+def normalize_design_location(
```

Comment:
this `normalize_design_location` function does nothing more than `varLib.models.normalizeLocation`